pennylane/pauli/grouping/group_observables.py

```diff
@@ -119,75 +152,326 @@ def binary_repr(self, n_qubits=None, wire_map=None):
+        ``self.complement_graph`` using a specified strategy from ``RX_STRATEGIES``. It then groups the indices
+        (nodes) of the graph by their assigned colours.
+
+        Returns:
+            dict[int, list[int]]: A dictionary where the keys are colours (integers) and the values are lists
+                of indices (nodes) that have been assigned that colour.
+        """
+        # A dictionary where keys are node indices and the value is the colour
+        if new_rx:
+            # 'strategy' kwarg was implemented in Rustworkx 0.15
+            colouring_dict = rx.graph_greedy_color(
+                self.complement_graph, strategy=RX_STRATEGIES[self.graph_colourer]
             )
-            mat_prod = (
-                self.binary_observables @ symplectic_form @ np.transpose(self.binary_observables)
+        else:
+            # Default value for <0.15.0 was 'lf'.
+            colouring_dict = rx.graph_greedy_color(self.complement_graph)
+
+        # group together indices (values) of the same colour (keys)
+        groups = defaultdict(list)
+        for idx, colour in sorted(colouring_dict.items()):
+            groups[colour].append(idx)
+
+        return groups
+
+    def idx_partitions_from_graph(self, observables_indices=None) -> tuple[tuple[int, ...], ...]:
+        """Use ``Rustworkx`` graph colouring algorithms to partition the indices of the Pauli observables into
+        tuples containing the indices of observables satisying the binary relation determined by ``self.grouping_type``.
+
+        Args:
+            observables_indices (tensor_like, optional): A tensor or list of indices associated to each observable.
+                This argument is helpful when the observables used in the graph colouring are part of a bigger set of observables.
+                Defaults to None. If ``None``, the partitions are made up of the relative indices, i.e. assuming ``self.observables``
+                have indices in [0, len(observables)-1].
+
+        Raises:
+            IndexError: When the tensor_like of observables_indices is not of the same length as the observables.
+
+        Returns:
+            tuple[tuple[int]]: Tuple of tuples containing the indices of the partitioned observables.
+        """
+        if observables_indices is None:
+            partition_indices = tuple(
+                tuple(indices) for indices in self._idx_partitions_dict_from_graph.values()
             )
+        else:
+            if len(observables_indices) != len(self.observables):
+                raise ValueError(
+                    f"The length of the list of indices: {len(observables_indices)} does not "
+                    f"match the length of the list of observables: {len(self.observables)}. "
+                )
+            partition_indices = self._partition_custom_indices(observables_indices)
 
-            if self.grouping_type == "commuting":
-                adj = mat_prod % 2
+        return partition_indices
```

Comment:
```suggestion
        if observables_indices is None:
             return tuple(
                 tuple(indices) for indices in self._idx_partitions_dict_from_graph.values()
             )
         if len(observables_indices) != len(self.observables):
             raise ValueError(
                 f"The length of the list of indices: {len(observables_indices)} does not "
                 f"match the length of the list of observables: {len(self.observables)}. "
             )
         return self._partition_custom_indices(observables_indices)
```

Similar to the stylistic suggestion above.